# BIVARIATE DATA ANALYSIS OF ILPD DATASET 

<hr>

[LINK TO ILPD DATASET](https://archive.ics.uci.edu/dataset/225/ilpd+indian+liver+patient+dataset)

Now , uploading the <b>dataset file</b> and
<b>importing necessary libraries.</b>

In [ ]:
%mamba install pandas
import pandas as pd

import matplotlib.pyplot as plt
df = pd.read_csv("Indian Liver Patient Dataset (ILPD).csv", header=None)

df.columns = [
    'Age', 'Gender', 'Total Bilirubin', 'Direct Bilirubin', 
    'Alkaline Phosphotase', 'Alamine Aminotransferase', 
    'Aspartate Aminotransferase', 'Total Proteins', 'Albumin', 
    'Albumin and Globulin Ratio', 'Dataset']

Now , making

# THE PLOT OF THE CORELATION MATRICES : 
<hr>

# PEARSON CORELATION COEFFICIENT MATRIX :

In [28]:
df['Gender-Number'] = df['Gender'].map({'Female': 0, 'Male': 1})
numeric_df = df.select_dtypes(include=['number'])

pearson_matrix = numeric_df.corr(method='pearson')
pearson_matrix

,Age,Total Bilirubin,Direct Bilirubin,Alkaline Phosphotase,Alamine Aminotransferase,Aspartate Aminotransferase,Total Proteins,Albumin,Albumin and Globulin Ratio,Dataset,Gender-Number
Age,1.000000,0.011763,0.007529,0.080425,-0.086883,-0.019910,-0.187461,-0.265924,-0.216408,-0.137351,0.056560
Total Bilirubin,0.011763,1.000000,0.874618,0.206669,0.214065,0.237831,-0.008099,-0.222250,-0.206267,-0.220208,0.089291
Direct Bilirubin,0.007529,0.874618,1.000000,0.234939,0.233894,0.257544,-0.000139,-0.228531,-0.200125,-0.246046,0.100436
Alkaline Phosphotase,0.080425,0.206669,0.234939,1.000000,0.125680,0.167196,-0.028514,-0.165453,-0.234166,-0.184866,-0.027496
Alamine Aminotransferase,-0.086883,0.214065,0.233894,0.125680,1.000000,0.791966,-0.042518,-0.029742,-0.002375,-0.163416,0.082332
Aspartate Aminotransferase,-0.019910,0.237831,0.257544,0.167196,0.791966,1.000000,-0.025645,-0.085290,-0.070040,-0.151934,0.080336
Total Proteins,-0.187461,-0.008099,-0.000139,-0.028514,-0.042518,-0.025645,1.000000,0.784053,0.234887,0.035008,-0.089121
Albumin,-0.265924,-0.222250,-0.228531,-0.165453,-0.029742,-0.085290,0.784053,1.000000,0.689632,0.161388,-0.093799
Albumin and Globulin Ratio,-0.216408,-0.206267,-0.200125,-0.234166,-0.002375,-0.070040,0.234887,0.689632,1.000000,0.163131,-0.003424
Dataset,-0.137351,-0.220208,-0.246046,-0.184866,-0.163416,-0.151934,0.035008,0.161388,0.163131,1.000000,-0.082416


<hr>

# SPEARMAN CORRELATION COEFFICIENT MATRIX :

In [29]:
spearman_matrix = numeric_df.corr(method='spearman')
spearman_matrix

,Age,Total Bilirubin,Direct Bilirubin,Alkaline Phosphotase,Alamine Aminotransferase,Aspartate Aminotransferase,Total Proteins,Albumin,Albumin and Globulin Ratio,Dataset,Gender-Number
Age,1.000000,0.113827,0.106473,0.059205,-0.067737,-0.018285,-0.174271,-0.260791,-0.249951,-0.129572,0.062364
Total Bilirubin,0.113827,1.000000,0.959216,0.383794,0.436586,0.508869,-0.019252,-0.222184,-0.285458,-0.303879,0.200503
Direct Bilirubin,0.106473,0.959216,1.000000,0.367818,0.412322,0.504138,-0.019987,-0.232664,-0.298187,-0.297270,0.209210
Alkaline Phosphotase,0.059205,0.383794,0.367818,1.000000,0.410752,0.395732,0.014028,-0.170809,-0.321725,-0.273247,0.079131
Alamine Aminotransferase,-0.067737,0.436586,0.412322,0.410752,1.000000,0.773611,-0.018811,-0.052673,-0.083335,-0.290709,0.201107
Aspartate Aminotransferase,-0.018285,0.508869,0.504138,0.395732,0.773611,1.000000,-0.084779,-0.204867,-0.209082,-0.308897,0.209434
Total Proteins,-0.174271,-0.019252,-0.019987,0.014028,-0.018811,-0.084779,1.000000,0.779077,0.273186,0.032220,-0.090905
Albumin,-0.260791,-0.222184,-0.232664,-0.170809,-0.052673,-0.204867,0.779077,1.000000,0.754140,0.167079,-0.095440
Albumin and Globulin Ratio,-0.249951,-0.285458,-0.298187,-0.321725,-0.083335,-0.209082,0.273186,0.754140,1.000000,0.188564,-0.008746
Dataset,-0.129572,-0.303879,-0.297270,-0.273247,-0.290709,-0.308897,0.032220,0.167079,0.188564,1.000000,-0.082416


<hr>

# ANALYSIS : 
<hr>

<b><h3>1. Ignore Direct Bilirubin.
Keep Total Bilirubin.</h3></b>

Total Bilirubin & Direct Bilirubin (Pearson: 0.87 | Spearman: 0.96)

Reason: 
They are highly correlated.
Also, Direct Bilirubin is a fraction of Total Bilirubin,
<hr>

<b><h3>2. Ignore ALT. Keep AST Liver Enzyme.</h3><b>

AST & ALT Liver Enzymes (Pearson: 0.79 | Spearman: 0.77)

Reason: They have a strong positive linear relationship. When the liver is damaged, both of these leak into the blood together.
    <hr>

<h3><b> 3. Ignore Albumin . Keep Total Proteins. </b></h3>

Albumin & Total Proteins (Pearson: 0.78 | Spearman: 0.78)

Reason: They have a strong positive linear relationship. Albumin makes up the vast majority of Total Proteins, and is thus included in total number of proteins.

<hr>

<b><h3> 4. OTHER HEALTH ATTRUBUTES SHOULD BE KEPT. </b></h3>

Reason :

<b> THEY SHOW STATISTICAL INDEPENDENCE. AS THEIR PEARSON AND SPEARMAN COEFFICIENTS ARE < 0.7 . </b>
<hr>